lib/matplotlib/axes/_axes.py

```diff
@@ -4319,14 +4319,14 @@ def do_patch(xs, ys, **kwargs):
                 do_box = do_patch if patch_artist else do_plot
                 boxes.append(do_box(box_x, box_y, **box_kw))
             # draw the whiskers
-            whiskers.append(do_plot(whis_x, whislo_y, **whisker_kw))
-            whiskers.append(do_plot(whis_x, whishi_y, **whisker_kw))
+            whiskers.append(do_plot(whis_x, whislo_y, **whisker_kw, label="_nolegend_"))
```

Comment:
This should be done as 

```
whisker_kw.setdefault('label', '_nolegend_)
```

or similar where we generate/accept these dictionaries.  As proposed this will fail with a TypeError for any user who is currently passing label though.